vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListFragment.kt

```diff
@@ -180,6 +183,14 @@ class HomeRoomListFragment @Inject constructor(
                     }
                 }.adapter
             }
+            is HomeRoomSection.RecentRoomsData -> RecentRoomCarouselController(
+                    avatarRenderer = avatarRenderer
+            ).also { controller ->
+                controller.listener = this
```

Comment:
I don't see any clean up of `RecyclerViews` in `onDestroyView()` of the fragment. Could we override it and clean the adapters associated to the lists (there is a `RecyclerView.cleanUp()` extension) + removing the listener from the controllers as we are passing the Fragment instance here.